lib/types.nix

```diff
@@ -373,6 +373,14 @@ rec {
       merge = mergeEqualOption;
     };
 
+    hex = mkOptionType {
+      name = "hex";
+      description = "hexadecimal number";
+      descriptionClass = "noun";
+      check = x: str.check x && builtins.match "[0-9a-fA-F]+" x != null;
+      merge = mergeEqualOption;
```

Comment:
I think it'd be good to canonicalize the notation, ideally before "merging".
We generally use lower case in the Nix ecosystem.
`0x` and `#` could be stripped (if allowed in the first place).
Leading zeroes should probably not be stripped, because the hex may be data instead of a number.

Reply:
> We generally use lower case in the Nix ecosystem.

I'll remove the matching for uppercase letters

> 0x and # could be stripped (if allowed in the first place).

I wouldn't allow them in the first place, not sure what other people think.

> Leading zeroes should probably not be stripped, because the hex may be data instead of a number.

that's not happening as far as I'm aware.

Reply:
> that's not happening as far as I'm aware.

:+1: and that's the right behavior for hex data (as opposed to a number, which wasn't clear before)